src/shared/nt-core/App.react.js

```diff
@@ -1,48 +1,30 @@
-import gql from 'graphql-tag'
 import PropTypes from 'prop-types'
 import React from 'react'
-import { graphql } from 'react-apollo'
-import { withRouter } from 'react-router'
-import { Route, Switch } from 'react-router-dom'
-import { compose } from 'recompose'
+import { Redirect, Route, Switch } from 'react-router-dom'
 
-import LoginPage from '../nt-login/LoginPage.react'
+import ProductListPage from '../nt-products/ProductListPage.react'
 import MainPage from './MainPage.react'
 import NoMatch from './NoMatch.react'
 
-const MainPageContainer = ({ match, data }) => data.product ? (
-  <MainPage match={match} product={data.product} />
-) : (
-  <NoMatch />
+const MainRouter = ({ component: Component, computedMatch, ...rest }) => (
+  <Route {...Object.assign(computedMatch, rest)} render={props =>
+    (computedMatch.params.productId && computedMatch.params.productId.match(/^[0-9a-fA-F]{24}$/)) ? (
```

Comment:
Valid `productId` doesn't mean it is existed in the database. Correct format of `productId` can still make an error if it doesn't exist in the database.